### PR TITLE
#157110627 Change how users adds meals

### PR DESCRIPTION
**Pull request by paulrimiru**
#### What does this PR do?
This PR changes the way a user adds meals to their nutrition plan
#### Description of Task to be completed?
When the user tries to make a nutrition plan, it is a bit uncomfortable to add a time for the meal and then have to add the food at that meal.
It will be better to set everything at once (time, ingredient and amount).
#### How should this be manually tested?
After cloning the repo. go to nutrition add nutrition plan then click on add new meal
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#157110627 change how users add meals](https://www.pivotaltracker.com/story/show/157110627)
###### User first adds a new meal
step one

###### User then adds meal items
step two

###### User performs both operations on one page
 
combined

#### Questions:
N/A

### CODE REVIEW

File: wger/nutrition/views/meal_item.py [hunk 1]
```diff
@@ -56,17 +56,24 @@ class MealItemCreateView(WgerFormMixin, CreateView):
...
         '''
-        meal = get_object_or_404(Meal, pk=kwargs['meal_id'])
-        if meal.plan.user == request.user:
-            self.meal = meal
-            return super(MealItemCreateView, self).dispatch(request, *args, **kwargs)
+        self.meal_id = kwargs.get('meal_id')
+        self.plan_pk = kwargs.get('plan_pk')
+        if self.meal_id:
+            meal = get_object_or_404(Meal, pk=self.meal_id)
+            if meal.plan.user == request.user:
```
Comment (Rodgers-M) [inline wger/nutrition/views/meal_item.py:70]: good job ensuring a user does not alter a meal they don't own

File: wger/utils/TimeWidget.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    """
+    A Widget that splits time input into <select> elements.
+    Allows form to show as 24hr: <hour>:<minute>:<second>, (default)
+    or as 12hr: <hour>:<minute>:<second> <am|pm> 
+    
+    Also allows user-defined increments for minutes/seconds
+    """
+    hour_field = '%s_hour'
+    minute_field = '%s_minute'
+    second_field = '%s_second'
```
Comment (Rodgers-M) [inline wger/utils/TimeWidget.py:32]: nice implementation of the time widget. Can the `second_field` be eliminated? Can a user be able to observe the seconds also?

**Review by ghost — COMMENTED** [review]
LGTM

File: wger/utils/TimeWidget.py (hunk 2 above)
Comment (paulrimiru) [reply wger/utils/TimeWidget.py:32]: I think it can be eliminated but it is set to 0 by default unless the user wants to change it.

**Comment by MarloneA** [comment]
Good work, I like that you improved the time input field and made it more intuitive

**Pull request merged**